sklearn/impute/tests/test_common.py

```diff
@@ -89,6 +90,10 @@ def test_imputers_add_indicator_sparse(imputer, marker):
     imputer.set_params(missing_values=marker, add_indicator=True)
 
     X_trans = imputer.fit_transform(X)
+    # scipy.sparse.hstack in the imputer transform will cause the output to become
+    # coo_matrix if there is no fast path, and coo_matrix does not support slicing
```

Comment:
Actually this is an issue because our user will face a change of behaviour. We need to make a fix such that the `X_trans` has the same format with arrays and matrices.